AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerServiceImpl.kt

```diff
@@ -204,6 +181,61 @@ internal class TestRunnerServiceImpl internal constructor(
         }
     }
 
+    private suspend fun findLogcats(
+        testMatrix: TestMatrix
+    ): MutableMap<TestRunnerService.TestIdentifier, TestRunnerService.ResultFileResource> {
+        var testCaseLogcatBlobs = mutableMapOf<TestRunnerService.TestIdentifier, TestRunnerService.ResultFileResource>()
+        val resultPath = testMatrix.resultStorage.googleCloudStorage.gcsPath
+
+        // get cached result if available, else invoke tool results apis to get test execution steps and store them in glcoud bucket
+        val logcatsFilePath = "$resultPath/logcats"
+        val logcatsRelativePath = logcatsFilePath.substringAfter("gs://").substringAfter("/")
+        if (googleCloudApi.existingFilePath(logcatsRelativePath) != null) {
+            googleCloudApi.walkEntries(GcsPath(logcatsFilePath)).first().obtainInputStream().reader().readLines().forEach {
+                val className = it.split("#")[0]
```

Comment:
i think it is probably better to only cache failed tests and then use 1 meta object to keep it.

e.g. on GCP, you would keep 1 json file per run that would include

```
data class FailedLogcats(val map<TestIdentifier, GcsLogcatPath>)
```

then just deserialize it from json here